### PR TITLE
minor: Optimize method distance in checkstyle.api.LocalizedMessage

### PR DESCRIPTION
**Pull request by wilcoln**
Before Optimization:
https://wilcoln.github.io/com.pupppycrawl.tools.checkstyle.api/before.LocalizedMessage.java.html

After Optimization:
https://wilcoln.github.io/com.pupppycrawl.tools.checkstyle.api/after.LocalizedMessage.java.html

Optimization was made manually and with respect to the currently hardcoded criteria defined in the method-distance repository.

### CODE REVIEW

**Comment by rdiachenko** [comment]
@wilcoln could you try to move `equals` and `hashCode` methods closer to `compareTo`, so the 'override' group of methods is not split.

**Comment by wilcoln** [comment]
> @wilcoln could you try to move `equals` and `hashCode` methods closer to `compareTo`, so the 'override' group of methods is not split.

Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java [hunk 1]
```diff
@@ -461,10 +365,49 @@ public static void setLocale(Locale locale) {
...
     }
 
+    /** Clears the cache. */
+    public static void clearCache() {
+        BUNDLE_CACHE.clear();
+    }
+
     ////////////////////////////////////////////////////////////////////////////
     // Interface Comparable methods
     ////////////////////////////////////////////////////////////////////////////
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:375]: Should be just before `public int compareTo(LocalizedMessage other)`

Comment (wilcoln) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/LocalizedMessage.java:375]: Done.

**Pull request merged**